ldap_check.sh

```diff
@@ -0,0 +1,300 @@
+	then
+		echo "ldapsearch is not in your path or not installed.  Please correct and rerun this script";
+		exit 1;
+fi
+
+
+# Check account vars have been entered
+
+check_ldap_config() {
+
+if [ -z "$port" ]
+        then
+                echo "Port Undefined, using default 636";
+                port=636;
+                read -rsp $'press key to continue \n' -n1 key
+fi
+
+if [ "$port" = "389 -ZZ" ]
+        then
+                uri=ldap
+        else
+                uri=ldaps
+fi
+
+if [ -z "$user" ]
+        then
+                echo -e "\nAccount variables undefined, enter information and reselect\n";
+                read_account;
+fi
+}
+
+# Define ldapsearch 
+
+ldsearch() {
+
+check_ldap_config
+
+$ldapsearch -H ${uri}://ldap.jumpcloud.com:${port} -x -b "ou=Users,o=${oid},dc=jumpcloud,dc=com" -D "uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com" -w "${pass}" "(objectClass=${search_param})" | less
+
+}
+
+#
+# Display common configuration strings
+#
+
+echo_ldap_config() {
+
+echo "###################################################################"
+echo "#                Common LDAP Configuration Settings               #"
+echo "###################################################################"
+echo ""
+echo "### URI/LDAP Server ###"
+echo ""
+echo "ldaps://ldap.jumpcloud.com:636"
+echo "ldap://ldap.jumpcloud.com:389"
+echo ""
+echo "### BIND DN ###"
+echo ""
+echo "uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com"
+echo ""
+echo "### Search DN/Base Search DN ###"
+echo ""
+echo "ou=Users,o=${oid},dc=jumpcloud,dc=com"
+echo ""
+echo "### LDAP Search Example ###"
+echo ""
+echo "$ldapsearch -H ${uri}://ldap.jumpcloud.com:${port} -x -b \"ou=Users,o=${oid},dc=jumpcloud,dc=com\" -D \"uid=${user},ou=Users,o=${oid},dc=jumpcloud,dc=com\" -w \"${pass}\" \"(objectClass=inetOrgPerson)\""
+echo ""
+echo "**If the above ldapsearch command results in invalid credentials, and the password contains special characters (!,@,#,etc...), replace the double quotes (\") around the password with single quotes (') and retry"
+echo ""
+}
+
+# Display the settings
+
+ldap_config() {
+
+check_ldap_config
+echo_ldap_config | less
+
+}
+
+#
+# read in oid, ldap service account and password
+#
+
+# read username, check for null input
+
+read_user() {
+echo -n "Enter LDAP Binding User account name: "
+        read user;
+if [ -z "$user" ]
+        then
+                echo "Input cannot be null"; read_user;
+        else
+                read_pass;
+fi
+}
+
+# read password, check for null input
+
+read_pass() {
+echo -n "Enter password: "
+        read -s pass;
+echo
+if [ -z "$pass" ]
+        then
+                echo "Input cannot be null"; read_pass;
+        else
+                read_oid;
+fi
+}
+
+# read oid, check for null input
+
+read_oid() {
+echo -n "Enter the organization ID: "
+        read oid;
+if [ -z "$oid" ]
+        then
+                echo "Input cannot be null"; read_oid;
+        else continue;
+fi
+}
+
+read_account() {
```

Comment:
This could turn into:
```
user=`readLine "LDAP Binding User account name"`
pass=`readLine "LDAP Binding User password"`
oid=`readLine "Organization ID"`
```
if you had one readLine function as mentioned above.